refinery/core/models.py

```diff
@@ -2213,3 +2220,87 @@ class SiteProfile(models.Model):
 
     def __unicode__(self):
         return self.site.name
+
+
+class SiteStatistics(models.Model):
+    """
+    A model to encapsulate various information about a deployed Refinery
+    Instance's utilization.
+    """
+    datasets_shared = models.IntegerField()
+    datasets_uploaded = models.IntegerField()
+    groups_created = models.IntegerField()
+    run_date = models.DateTimeField()
+    total_user_logins = models.IntegerField()
+    total_visualization_launches = models.IntegerField()
+    total_workflow_launches = models.IntegerField()
+    users_created = models.IntegerField()
+    unique_user_logins = models.IntegerField()
+
+    class Meta:
+        verbose_name_plural = "site statistics"
+
+    def save(self, force_insert=False, force_update=False, using=None,
+             update_fields=None):
+        self.run_date = timezone.now()
+        self.datasets_uploaded = self._get_datasets_uploaded().count()
+        self.datasets_shared = len(self._get_datasets_shared())
+        self.groups_created = self._get_number_of_groups_created()
+        self.users_created = self._get_users_created().count()
+        self.unique_user_logins = self._get_unique_user_logins().count()
+        self.total_user_logins = self._get_number_of_total_user_logins()
+        self.total_workflow_launches = \
+            self._get_total_workflow_launches().count()
+        self.total_visualization_launches = \
+            self._get_total_visualization_launches().count()
+        super(SiteStatistics, self).save()
+
+    def _get_previous_instance(self):
+        previous_instance = SiteStatistics.objects.filter(
+            run_date__lt=self.run_date
+        ).order_by('-run_date').first()
+
+        # Return the previous instance by run_date, or the current instance
+        # if a prior one doesn't exist
+        return previous_instance or self
+
+    def _get_datasets_shared(self):
+        return [
+            dataset for dataset in DataSet.objects.filter(
+                modification_date__gte=self._get_previous_instance().run_date
+            ) if dataset.shared
+        ]
+
+    def _get_datasets_uploaded(self):
+        return DataSet.objects.filter(
+            creation_date__gte=self._get_previous_instance().run_date
+        )
+
+    def _get_total_visualization_launches(self):
+        return tool_manager.models.VisualizationTool.objects.filter(
+            creation_date__gte=self._get_previous_instance().run_date
+        )
```

Comment:
Return just the number of tools instead of the queryset?

Reply:
I left this method not returning a count so that it would be easier to extend in the future when we want to know about the objects created rather than just their counts